packages/config/config/devices/0x0190/adc-t_3000.json

```diff
@@ -0,0 +1,804 @@
+					"value": 0
+				},
+				{
+					"label": "Enabled",
+					"value": 1
+				}
+			]
+		},
+		{
+			"#": "43",
+			"label": "Outdoor Temperature Timeout",
+			"unit": "minutes",
+			"valueSize": 2,
+			"minValue": 0,
+			"maxValue": 32767,
+			"defaultValue": 1440,
+			"unsigned": true
+		},
+		{
+			"#": "45",
+			"label": "Heat Pump Expire",
+			"unit": "minutes",
+			"valueSize": 2,
+			"minValue": 0,
+			"maxValue": 2880,
+			"defaultValue": 0
+		},
+		{
+			"#": "46",
+			"label": "Dehumidify by AC Offset",
+			"unit": "°F",
+			"valueSize": 4,
+			"minValue": 0,
+			"maxValue": 10,
+			"defaultValue": 3
+		},
+		{
+			"#": "48",
+			"label": "PIR Enable",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 1,
+			"defaultValue": 1,
+			"allowManualEntry": false,
+			"options": [
+				{
+					"label": "Disabled",
+					"value": 0
+				},
+				{
+					"label": "Enabled",
+					"value": 1
+				}
+			]
+		},
+		{
+			"#": "49",
+			"label": "Humidity Display",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 1,
+			"defaultValue": 1,
+			"allowManualEntry": false,
+			"options": [
+				{
+					"label": "Disabled",
+					"value": 0
+				},
+				{
+					"label": "Enabled",
+					"value": 1
+				}
+			]
+		},
+		{
+			"#": "50",
+			"label": "System configuration",
+			"description": "Summarized report of system configuration",
+			"valueSize": 4,
+			"minValue": 0,
+			"maxValue": 4294967295,
+			"unsigned": true,
+			"readOnly": true
+		},
+		{
+			"#": "51",
+			"label": "Thermostat Reset",
+			"description": "Must write the magic value 2870 to take effect.",
+			"valueSize": 2,
+			"minValue": 0,
+			"maxValue": 2870,
+			"defaultValue": 0,
+			"writeOnly": true
+		},
+		{
+			"#": "52",
+			"label": "Vent Options",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 4,
+			"defaultValue": 4,
+			"allowManualEntry": false,
+			"options": [
+				{
+					"label": "Disabled",
+					"value": 0
+				},
+				{
+					"label": "Always activate regardless of thermostat operating state",
+					"value": 1
+				},
+				{
+					"label": "Only activate when heating",
+					"value": 2
+				},
+				{
+					"label": "Only activate when cooling",
+					"value": 3
+				},
+				{
+					"label": "Only activate when heating or cooling",
+					"value": 4
+				}
+			]
+		},
+		{
+			"#": "53",
+			"label": "Vent Circulation Period",
+			"unit": "minutes",
+			"valueSize": 2,
+			"minValue": 10,
+			"maxValue": 1440,
+			"defaultValue": 60
+		},
+		{
+			"#": "54",
+			"label": "Vent Circulation Duty Cycle",
+			"unit": "%",
+			"valueSize": 1,
+			"minValue": 0,
+			"maxValue": 100,
+			"defaultValue": 25
+		},
+		{
+			"#": "55",
+			"label": "Vent Maximum Outdoor Temperature",
+			"unit": "°F",
+			"valueSize": 4,
+			"minValue": 0,
+			"maxValue": 99,
+			"defaultValue": 99
+		},
+		{
+			"#": "56",
+			"label": "Vent Minimum Outdoor Temperature",
+			"unit": "°F",
+			"valueSize": 4,
+			"minValue": 0,
+			"maxValue": 99,
+			"defaultValue": 0
```

Comment:
the default for 55 and 56 is technically 0xFFFF which disables this functionality. But since this is outside the min/max range, I didn't think this would work. Is it possible to specify an `options` array with values that are outside the allowed range?

Reply:
We typically mention it in the description in this case and just widen the min/max range:
```
"description": "Allowable range: 0-99.",
```
Then you can add an option for "Disable".

Reply:
@EiNSTeiN- Are these really 4 bytes wide but use 0xffff (2 bytes) for disabling?
If they were two bytes, you could make the parameter signed and set the range -1..99 where -1 is disable.

But on the other hand, manufacturer choices don't always make sense...

Reply:
per the manual, all the temperature values are 4 bytes, where the first byte is precision/scale/size. In practice, precision is always 1, scale is always Fahrenheit, and size is always 2. Byte 2-3 is the temperature, which is why 0xFFFF (2 bytes) is the disabled magic value. Also since precision is 1, a range of 0-99 F is actually represented as 0-990. Some temperature offsets can be negative as well.

I could tighten the ranges if that's what you are asking. I initially thought the precision could be changed from 1, which would change the range, but from what I tested, that's not possible.